csharp/src/Google.Protobuf/Reflection/FileDescriptor.cs

```diff
@@ -63,15 +63,15 @@ static FileDescriptor()
 
         private readonly Lazy<Dictionary<IDescriptor, DescriptorDeclaration>> declarations;
 
-        private FileDescriptor(ByteString descriptorData, FileDescriptorProto proto, IEnumerable<FileDescriptor> dependencies, DescriptorPool pool, bool allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
+        private FileDescriptor(ByteString descriptorData, FileDescriptorProto proto, IList<FileDescriptor> dependencies, DescriptorPool pool, bool allowUnknownDependencies, GeneratedClrTypeInfo generatedCodeInfo)
         {
             SerializedData = descriptorData;
             DescriptorPool = pool;
             Proto = proto;
             // Note: the Edition property relies on the proto being set first, so this line
             // has to come after Proto = proto.
             Features = FeatureSetDescriptor.GetEditionDefaults(Edition).MergedWith(proto.Options?.Features);
-            Dependencies = new ReadOnlyCollection<FileDescriptor>(dependencies.ToList());
+            Dependencies = new ReadOnlyCollection<FileDescriptor>(dependencies);
```

Comment:
I disagree with this change. It means that if the dependencies have been passed in externally (as they are from FromGeneratedCode) we don't end up making a defensive copy, and the collection could be modified. Yes, often this means we make a defensive copy unnecessarily - but I'd personally rather have that than have weird potential effects due to mutation.

Reply:
Yeah, I can see that it is obsoleted as well.
I have reverted the removal of the copy